app/code/Magento/Rule/Test/Unit/Model/Condition/AbstractConditionTest.php

```diff
@@ -65,6 +65,9 @@ public static function validateAttributeDataProvider()
             [null, '==', 0, false],
             [null, '==', 0.00, false],
 
+            ['0123', '===', '123', false],
```

Comment:
In addition of the above test cases, shall we add more test cases as below to cover wide range of scenarios:

```
// Test cases for strict equality with leading zeros
['000123', '===', '123', false],
['123', '===', '000123', false],
['0123', '===', '0123', true]

// Test cases for strict equality with different numeric types
[123, '===', '123', false],
['123', '===', 123, false]
```